apps/dashboard/src/contract-ui/hooks/useContractSources.ts

```diff
@@ -1,17 +1,25 @@
-import { useQueryWithNetwork } from "@3rdweb-sdk/react/hooks/query/useQueryWithNetwork";
-import { useSDK } from "@thirdweb-dev/react";
+import { useQuery } from "@tanstack/react-query";
+import type { ThirdwebContract } from "thirdweb";
+import { getCompilerMetadata } from "thirdweb/contract";
+import invariant from "tiny-invariant";
 
-export function useContractSources(contractAddress?: string) {
-  const sdk = useSDK();
-  return useQueryWithNetwork(
-    ["contract-sources", contractAddress],
-    async () => {
-      return sdk
-        ?.getPublisher()
-        .fetchContractSourcesFromAddress(contractAddress || "");
+export function useContractSources(contract?: ThirdwebContract) {
+  return useQuery({
+    queryKey: [
+      "contract-sources",
+      contract?.chain.id || "",
+      contract?.address || "",
+    ],
+    queryFn: async (): Promise<Array<{ filename: string; source: string }>> => {
+      invariant(contract, "contract is required");
+      const data = await getCompilerMetadata(contract);
+      const sources = data.metadata.sources || {};
+      const arr = Object.keys(sources).map((key) => ({
+        filename: key,
+        source: sources[key]?.content || "",
```

Comment:
this is not enough to get the sources. if you look at the implementation in v4, we actually check if there's an url for the attached source (ipfs hash) and download the sources in there

Reply:
ah I see, ty ser!